### PR TITLE
bid slippage support

### PR DESCRIPTION
**Pull request by doingli**
1. The auction supports bidding a higher price. expectedSpending indicates the highest price that can be accepted.
2. The _bid method automatically approves. In this way, the web side does not need to call the approve method before each bid, and reducing the advantages of bot trading

### CODE REVIEW

File: contracts/nft/PFPAuction.sol [hunk 1]
```diff
@@ -207,6 +207,8 @@ contract PFPAuction is OwnableUpgradeable, ReentrancyGuardUpgradeable, IERC721Re
   }
 
   function bid(address tokenAddress, uint256 tokenId, uint256 expectedSpending) external payable nonReentrant {
+    uint256 price = getNextPrice(tokenAddress, tokenId);
+    _lfg.approve(address(this), price);
```
Comment (sullof) [inline contracts/nft/PFPAuction.sol:211]: What is this?
The contract cannot approve itself for spending users' LFG; with this command, it is approving itself to spend its own LFG, which I don't think is what you want.

**Review by sullof — CHANGES_REQUESTED** [review]
I don't think that that lfg.approve(address(this), price) makes any sense.
BTW, my original proposal for the PFPAuction was managing overbidding for both native and LFG bids and you asked me to remove it. You may just reintroduce it.
If you want the user to make a unique large approval, you can overbid, managing the expected spending as a maximum expected spending.

File: contracts/nft/PFPAuction.sol (hunk 1 above)
Comment (doingli) [reply contracts/nft/PFPAuction.sol:211]: Got it, I misunderstood the Approve method

**Comment by doingli** [comment]
Discard this version

**Pull request closed without merging**